### PR TITLE
`<BounceAnimation/>` and `<Animate/>` components refactor

### PR DESCRIPTION
**Pull request by TalZa1810**
### 🔦 Summary
Adding a `<BounceAnimation/>` component which uses `<Animate/>` component composition. 

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [x] Story
  - [x] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [x] Component
  - [ ] Visual test

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6426

File: src/BounceAnimation/BounceAnimation.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  /** Applied as data-hook HTML attribute that can be used in the tests */
+  dataHook: PropTypes.string,
+
+  /** The component which we would like to animate. When given multiple components- only the first child node will be animated. */
+  children: PropTypes.node.isRequired,
+
+  /** Triggers the animation transition */
+  active: PropTypes.bool,
+
+  /** A callback fired immediately after the transition starts.*/
```
Comment (gendeld) [inline src/BounceAnimation/BounceAnimation.js:73]:
```suggestion
  /** A callback fired immediately after the animation starts. */
```

File: src/BounceAnimation/BounceAnimation.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  onStart: PropTypes.func,
+
+  /** A callback fired immediately after the transition ended.*/
```
Comment (gendeld) [inline src/BounceAnimation/BounceAnimation.js:76]:
```suggestion
  /** A callback fired immediately after the animation ends. */
```

File: src/BounceAnimation/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import * as React from 'react';
+
+export interface BounceAnimationProps {
+  dataHook?: string;
+  children: React.ReactNode;
+  active?: boolean;
+  onStart?(): void;
```
Comment (gendeld) [inline src/BounceAnimation/index.d.ts:7]:
```suggestion
  onStart?: (event: React.AnimationEvent<HTMLDivElement>) => void;
```

File: src/BounceAnimation/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  onEnd?(): void;
```
Comment (gendeld) [inline src/BounceAnimation/index.d.ts:8]:
```suggestion
  onEnd?: (event: React.AnimationEvent<HTMLDivElement>) => void;
```

**Comment by gendeld** [comment]
https://github.com/wix/wix-style-react/blob/09473dbb705ac674511230da4bad17c07c2f8fcd/src/Animate/index.d.ts#L12
```
declare const Animate: React.FC<AnimateProps>;
export default Animate;
```

File: src/BounceAnimation/BounceAnimation.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  }
+}
+
+BounceAnimation.displayName = 'Bounce';
+
+BounceAnimation.propTypes = {
+  /** Applied as data-hook HTML attribute that can be used in the tests */
+  dataHook: PropTypes.string,
+
+  /** The component which we would like to animate. When given multiple components- only the first child node will be animated. */
```
Comment (TalZa1810) [inline src/BounceAnimation/BounceAnimation.js:67]:
```suggestion
  /** The component which we would like to animate. */
```

**Pull request merged**